src/box/rt_lua_call_access.c

```diff
@@ -0,0 +1,101 @@
+#include "rt_lua_call_access.h"
+#include "diag.h"
+#include "assoc.h"
+#include "user.h"
+#include "session.h"
+#include "../lua/init.h" /* tarantool_lua_is_builtin_global */
+
+static struct mh_access_t *user_rt_access_cache = NULL;
+static struct mh_strnptr_t *user_rt_universe_access_cache = NULL;
+
+const char *
+user_name_by_id(uint32_t uid, uint32_t *name_len)
+{
+	struct user *user = user_find(uid);
+	const char *name = user->def->name;
+	*name_len = strlen(name);
+	return name;
+}
+
+void
+rt_lua_call_access_init(void)
+{
+	user_rt_access_cache = mh_access_new();
+	user_rt_universe_access_cache = mh_strnptr_new();
+}
+
+void
+rt_lua_call_access_destroy(void)
+{
+	mh_access_delete(user_rt_access_cache);
+	mh_strnptr_delete(user_rt_universe_access_cache);
+}
+
+void
+grant_rt_access(const char *uname, uint32_t uname_len,
+		const char *fname, uint32_t fname_len)
+{
+	/* Grant universe access. */
+	if (is_universe(fname, fname_len)) {
+		uint32_t hash = mh_strn_hash(uname, uname_len);
+		const struct mh_strnptr_node_t uid_node =
+			{ uname, uname_len, hash, NULL };
+		struct mh_strnptr_node_t repl_uid_node = { NULL, 0, 0, NULL };
+		struct mh_strnptr_node_t *prepl_uid_node = &repl_uid_node;
+		mh_strnptr_put(user_rt_universe_access_cache, &uid_node,
+			       &prepl_uid_node, NULL);
```

Comment:
May be simplified to:

```c
struct mh_strnptr_node_t *ignored = NULL;
mh_strnptr_put(<...>, &ignored, <...>);
```

And the same below.

Reply:
Done.